pandas/core/internals.py

```diff
@@ -1070,7 +1070,16 @@ def eval(self, func, other, raise_on_error=True, try_cast=False, mgr=None):
                                      "values [%s]" % (values.T.shape,
                                                       other.shape))
 
-        transf = (lambda x: x.T) if is_transposed else (lambda x: x)
+        def transf(x):
+            if is_transposed:
+                x = x.T
+
+            # fix issue #12564: CategoricalBlock is 1-dim only, while category
+            # datarame can be more.
+            if getattr(self, 'is_categorical', None) and x.ndim < self.ndim:
```

Comment:
these are too specific here for this type of general routine. you *might* want to do this in ``_try_coerce_results`` for only Categoricals.